api-server/src/main/java/io/enmasse/api/v1/http/HttpAddressSpaceService.java

```diff
@@ -119,6 +124,65 @@ public Response createAddressSpace(@Context SecurityContext securityContext, @Co
         });
     }
 
+    @PATCH
+    @Consumes({MediaType.APPLICATION_JSON_PATCH_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("{addressSpace}")
+    public Response patchAddressSpace(@Context SecurityContext securityContext,
+                                      @PathParam("namespace") String namespace,
+                                      @PathParam("addressSpace") String addressSpaceName,
+                                      @NotNull JsonPatch patch) throws Exception {
+
+        return doPatch(securityContext, namespace, addressSpaceName,
+                jsonNode -> patch.apply(jsonNode));
+    }
+
+    @PATCH
+    @Consumes({"application/merge-patch+json"})
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("{addressSpace}")
+    public Response patchAddressSpace(@Context SecurityContext securityContext,
+                                      @PathParam("namespace") String namespace,
+                                      @PathParam("addressSpace") String addressSpaceName,
+                                      @NotNull JsonMergePatch patch) throws Exception {
+
+
+        return doPatch(securityContext, namespace, addressSpaceName,
+                jsonNode -> patch.apply(jsonNode));
+    }
+
+    private Response doPatch(@Context SecurityContext securityContext, @PathParam("namespace") String namespace, @PathParam("addressSpace") String addressSpaceName,
+                             CheckedFunction<JsonNode, JsonNode, JsonPatchException> patcher) throws Exception {
+        return doRequest("Error patching address space " + addressSpaceName, () -> {
+            verifyAuthorized(securityContext, namespace, ResourceVerb.update);
```

Comment:
This should use the verb patch i think

Reply:
> This should use the verb patch i think

I don't understand the comment. Could you clarify? This private method is a shared utility method that is shared by the two public methods that are annotated @PATCH.

Reply:
I mean ResourceVerb.patch instead of ResourceVerb.update. Kubernetes uses patch in RBAC to authorize the patch operation.

See https://kubernetes.io/docs/reference/access-authn-authz/rbac/

Reply:
Ah, I see, yes I'll fix that.